miso-web/src/spring-test/java/uk/ac/bbsrc/tgac/miso/webapp/springtest/BoxRestControllerST.java

```diff
@@ -0,0 +1,284 @@
+    getMockMvc()
+        .perform(post(CONTROLLER_BASE + "/1/bulk-remove")
+            .content(makeJson(positions))
+            .contentType(MediaType.APPLICATION_JSON))
+        .andExpect(status().isOk());
+
+    BoxImpl box = currentSession().get(entityClass, 1);
+    assertTrue(box.isFreePosition("A07"));
+    assertTrue(box.isFreePosition("B05"));
+  }
+
+  @Test
+  public void testDiscardSingleItem() throws Exception {
+    getMockMvc()
+        .perform(post(CONTROLLER_BASE + "/1/positions/A07/discard"))
+        .andExpect(status().isOk());
+
+    BoxImpl box = currentSession().get(entityClass, 1);
+    assertTrue(box.isFreePosition("A07"));
+
+    SampleImpl sample = currentSession().get(SampleImpl.class, 205);
+    assertTrue(sample.isDiscarded());
+  }
+
+  @Test
+  public void testDiscardMultipleItems() throws Exception {
+    List<String> positions = Arrays.asList("A07", "B05");
+    getMockMvc()
+        .perform(post(CONTROLLER_BASE + "/1/bulk-discard")
+            .content(makeJson(positions))
+            .contentType(MediaType.APPLICATION_JSON))
+        .andExpect(status().isOk());
+
+    BoxImpl box = currentSession().get(entityClass, 1);
+    assertTrue(box.isFreePosition("A07"));
+    assertTrue(box.isFreePosition("B05"));
+
+    SampleImpl sample = currentSession().get(SampleImpl.class, 205);
+    assertTrue(sample.isDiscarded());
+    sample = currentSession().get(SampleImpl.class, 206);
+    assertTrue(sample.isDiscarded());
+  }
+
+  @Test
+  public void testDiscardEntireBox() throws Exception {
+    getMockMvc()
+        .perform(post(CONTROLLER_BASE + "/1/discard-all"))
+        .andExpect(status().isNoContent());
+
+    BoxImpl box = currentSession().get(entityClass, 1);
+    assertEquals(96, box.getFreeCount());
+  }
+
+  @Test
+  public void testBulkUpdatePositions() throws Exception {
+    BulkUpdateRequestItem item1 = new BulkUpdateRequestItem();
+    item1.setPosition("A07");
+    item1.setSearchString("SAM205");
+
+    BulkUpdateRequestItem item2 = new BulkUpdateRequestItem();
+    item2.setPosition("C06");
+    item2.setSearchString("SAM204");
+
+    List<BulkUpdateRequestItem> items = Arrays.asList(item1, item2);
+
+    getMockMvc()
+        .perform(post(CONTROLLER_BASE + "/1/bulk-update")
+            .content(makeJson(items))
+            .contentType(MediaType.APPLICATION_JSON))
+        .andExpect(status().isOk());
+
+    BoxImpl updatedBox = currentSession().get(entityClass, 1);
+    assertFalse(updatedBox.isFreePosition("A07"));
+    assertFalse(updatedBox.isFreePosition("C06"));
+  }
+
+  @Test
+  public void testSetBoxLocation() throws Exception {
+    getMockMvc()
+        .perform(post(CONTROLLER_BASE + "/1/setlocation")
+            .param("storageId", "7"))
+        .andExpect(status().isAccepted());
+
+    BoxImpl box = currentSession().get(entityClass, 1);
+    assertEquals("Loose Storage", box.getStorageLocation().getLocationUnit().getDisplayName());
```

Comment:
There could be many locations named this. You chose the location by ID above, so confirm the ID here